src/main/java/tigase/mix/model/DefaultMixLogic.java

```diff
@@ -233,6 +233,17 @@ public void checkPermission(BareJID channel, BareJID senderJid, MixAction action
 					}
 				}
 				break;
+			case retract:
+				ChannelConfiguration configuration3 = mixRepository.getChannelConfiguration(channel);
+				if (configuration3 != null) {
+					if (!configuration3.isUserMessageRetraction()) {
+						switch (configuration3.getAdministratorMessageRetractionRights()) {
+							
```

Comment:
empty switch block?

Reply:
Good catch. Initially, I wanted to use this method to test permission for retraction but in the end I've decided to implement check directly in `ChannelGroupChatMessageModule` and this code is not used. I'll clean it up.